### PR TITLE
fix: issues with semaphore and rln boilerplate

### LINKED ISSUE #96: Semaphore proof generation with approve step
For now there is an error with semaphore proof generation when we have approve modal.
Need to check and fix this user path.

Image

### LINKED ISSUE #98: Make user inputs read only for extension approve modal
We need to disable editing inputs for user on extension side.

Questions:

- [x] Do we allow user to change Signal and External nullifier in extension? update: no, we don't.
- [x] Do we allow user to enter random signal value? update: no, it's provided by external party.

Image

### PR DESCRIPTION
**Pull request by 0xmad**
## Explanation

Currently, there are some errors for semaphore proof generation with approve modal and some minor improvements.

Details are below:
- [x] Add fixes for lock service to track state and fire logouts
- [x] Minor fix for message push
- [x] Add autofocus on first input for onboarding
- [x] Prettier/linter fixes for related files
- [x] Make approve inputs read only
- [x] Add content script boilerplate for RLN proof generation
- [x] Support RLN generation on background script
- [x] Check if there is no active identity on first launch
- [x] Update demo website with unsubscribe functions

## More Information

Fixes #98
Fixes #96 

## Screenshots/Screencaps

N/A

## Manual Testing Steps

1. Install extension
2. Don't use approve for all the requests
3. Try to generate semaphore proof

## Pre-Merge Checklist

- [x] PR template is filled out
- [ ] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [ ] PR has been added to the appropriate release Milestone

### + If there are functional changes:

- [ ] Manual testing complete & passed
- [ ] "Extension QA Board" label has been applied

> PR template source from [github.com/MetaMask](https://github.com/MetaMask)

### CODE REVIEW

File: src/background/controllers/browser-utils.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ class BrowserUtils {
 
   openPopup = async () => {
     if (this.cached) {
-      this.focusWindow(this.cached.id);
+      await this.focusWindow(this.cached.id);
```
Comment (iskdrews) [inline src/background/controllers/browser-utils.ts:36]: Shouldn't we await for the `focusWindow()`?

**Review by iskdrews — CHANGES_REQUESTED** [review]
All looks good to me, just some small issues. Including the weird extension size issue.

File: src/background/controllers/browser-utils.ts (hunk 1 above)
Comment (0xmad) [reply src/background/controllers/browser-utils.ts:36]: Yes, I have await here.

File: src/background/zk-keeper.ts [hunk 2]
```diff
@@ -138,20 +128,21 @@ export default class ZkKeeperController extends Handler {
       async (payload: SemaphoreProofRequest, meta: any) => {
         const { unlocked } = await LockService.getStatus();
 
+        await BrowserUtils.openPopup();
```
[the remark this answers is not in the document]
Comment (0xmad) [reply src/background/zk-keeper.ts:131]: I'll fix it

**Review by iskdrews — APPROVED** [review]
Approved! thanks @0xmad <3

**Pull request merged**